### PR TITLE
SCRUM-7: implement POST /users

### PR DESCRIPTION
**Pull request by RickyLeeChi**
Create(/Post) user by username, password, email, phoneNumber.

- Just make simple rule for password to provide at least 8 to 15 character
- Valid phone format align with /PUT users
- Thanks @jasoncintao for providing PasswordUtils that used in /POST as well.
- Just writing a very simple spock unit test for UserService in creating user
- Add /POST API in Postman collection

TODO in the future PR: Implement full unit testing about /POST (creating user) for better unit testing coverage.

### CODE REVIEW

**Review by Hannibal-Tao — DISMISSED** [review]
Looks good

File: user-service/src/test/groovy/com/cdk/modern/renting/userservice/UserControllerTest.groovy [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        userRepository.save(_ as User) >> userDomain
+        roleRepository.findByName(_ as String) >> roleDomain
+
+        when:
+        UserInfoResponse result = userService.createUser(user)
+
+        then:
+        with(result) {
+            username: 123
```
Comment (Hannibal0416) [inline user-service/src/test/groovy/com/cdk/modern/renting/userservice/UserControllerTest.groovy]: here should be `username == "123"`

Comment (RickyLeeChi) [reply user-service/src/test/groovy/com/cdk/modern/renting/userservice/UserControllerTest.groovy]: Thanks for the catch.
Updated.

**Review by Hannibal0416 — APPROVED** [review]
Looks good

**Pull request merged**